invenio_rdm_records/services/config.py

```diff
@@ -104,7 +104,7 @@ def is_record_and_has_parent_doi(record, ctx):
 def has_doi(record, ctx):
     """Determine if a record has a DOI."""
     pids = record.pids or {}
-    return "doi" in pids
+    return "doi" in pids and pids["doi"].get("identifier")
```

Comment:
this change seems a bit not backwards compatible, even if it will be evaluated to true in normal comparison, if someone used `has_doi is True` or similar, it will explode or return unexpected behaviour
maybe we should consider using types at some point ...?